### PR TITLE
Project Enumerable Methods

### PR DESCRIPTION
**Pull request by jstiven01**

### CODE REVIEW

File: enumerable_methods.rb [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    if object.is_a?(Array)
+      object.length.times do |i|
+        yield (object[i])
+      end
+    elsif object.is_a?(Hash)
+      object.keys.length.times do |i|
+        yield [object.keys[i], object.values[i]]
+      end
+    end
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]: Should return an **enumerator**

```ruby
irb(main):010:0> array
=> [1, 7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):011:0> array.each
=> #<Enumerator: [1, 7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]:each>
irb(main):012:0> array.my_each
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):12
        1: from enumerable_methods.rb:5:in `my_each'
ArgumentError (wrong number of arguments (given 0, expected 1))
irb(main):013:0> 
```

File: enumerable_methods.rb [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    if object.is_a?(Array)
+      object.length.times do |i|
+        yield [object[i], i]
+      end
+    elsif object.is_a?(Hash)
+      object.keys.length.times do |i|
+        yield [[object.keys[i], object.values[i]], i]
+      end
+    end
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):017:0> array
=> [1, 7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):018:0> my_each_output = ''
=> ""
irb(main):019:0> block = proc { |num, idx| my_each_output += "Num: #{num}, idx: #{idx}\n" }
=> #<Proc:0x000055bf3ef402d8@(irb):19>
irb(main):020:0> array.each_with_index(&block)
=> [1, 7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):021:0> each_output = my_each_output.dup
=> "Num: 1, idx: 0\nNum: 7, idx: 1\nNum: 7, idx: 2\nNum: 0, idx: 3\nNum: 1, idx: 4\nNum: 1, idx: 5\nNum: 4, idx: 6\nNum: 7, idx: 7\nNum: 1, idx: 8\nNum: 6, idx: 9\nNum: 7, idx: 10\nNum: 6, idx: 11\nNum: 4, idx: 12\nNum: 6, idx: 13\nNum: 4, idx: 14\nNum: 1, idx: 15\nNum: 1, idx: 16\nNum: 3, idx: 17\nNum: 3, idx: 18\nNum: 6, idx: 19\nNum: 5, idx: 20\nNum: 5, idx: 21\nNum: 7, idx: 22\nNum: 1, idx: 23\nNum: 5, idx: 24\nNum: 0, idx: 25\nNum: 8, idx: 26\nNum: 1, idx: 27\nNum: 8, idx: 28\nNum: 3, idx: 29\nNum: 8, idx: 30\nNum: 0, idx: 31\nNum: 4, idx: 32\nNum: 2, idx: 33\nNum: 8, idx: 34\nNum: 2, idx: 35\nNum: 4, idx: 36\nNum: 7, idx: 37\nNum: 5, idx: 38\nNum: 1, idx: 39\nNum: 2, idx: 40\nNum: 3, idx: 41\nNum: 7, idx: 42\nNum: 7, idx: 43\nNum: 2, idx: 44\nNum: 1, idx: 45\nNum: 0, idx: 46\nNum: 6, idx: 47\nNum: 6, idx: 48\nNum: 3, idx: 49\nNum: 5, idx: 50\nNum: 4, idx: 51\nNum: 2, idx: 52\nNum: 8, idx: 53\nNum: 1, idx: 54\nNum: 3, idx: 55\nNum: 2, idx: 56\nNum: 4, idx: 57\nNum: 0, idx: 58\nNum: 5, idx: 59\nNum: 1, idx: 60\nNum: 3, idx: 61\nNum: 2, idx: 62\nNum: 4, idx: 63\nNum: 5, idx: 64\nNum: 6, idx: 65\nNum: 7, idx: 66\nNum: 5, idx: 67\nNum: 6, idx: 68\nNum: 2, idx: 69\nNum: 5, idx: 70\nNum: 7, idx: 71\nNum: 8, idx: 72\nNum: 2, idx: 73\nNum: 5, idx: 74\nNum: 8, idx: 75\nNum: 6, idx: 76\nNum: 6, idx: 77\nNum: 5, idx: 78\nNum: 4, idx: 79\nNum: 5, idx: 80\nNum: 4, idx: 81\nNum: 3, idx: 82\nNum: 2, idx: 83\nNum: 7, idx: 84\nNum: 2, idx: 85\nNum: 8, idx: 86\nNum: 8, idx: 87\nNum: 3, idx: 88\nNum: 3, idx: 89\nNum: 5, idx: 90\nNum: 1, idx: 91\nNum: 8, idx: 92\nNum: 5, idx: 93\nNum: 1, idx: 94\nNum: 0, idx: 95\nNum: 4, idx: 96\nNum: 0, idx: 97\nNum: 6, idx: 98\nNum: 8, idx: 99\n"
irb(main):022:0> my_each_output = ''
=> ""
irb(main):023:0> array.my_each_with_index(&block)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):23
        1: from enumerable_methods.rb:17:in `my_each_with_index'
ArgumentError (wrong number of arguments (given 0, expected 1))
irb(main):024:0>
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):026:0> array
=> [1, 7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):027:0> array.each_with_index
=> #<Enumerator: [1, 7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]:each_with_index>
irb(main):028:0> array.my_each_with_index
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):28
        1: from enumerable_methods.rb:17:in `my_each_with_index'
ArgumentError (wrong number of arguments (given 0, expected 1))
irb(main):029:0> 
```

File: enumerable_methods.rb [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    elsif object.is_a?(Hash)
+      new_object = {}
+      my_each(object) do |key, value|
+        next unless yield [key, value]
+
+        new_object[key] = value
+      end
+    end
+    new_object
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):031:0> array
=> [1, 7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):032:0> array.select
=> #<Enumerator: [1, 7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]:select>
irb(main):033:0> array.my_select
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):33
        1: from enumerable_methods.rb:29:in `my_select'
ArgumentError (wrong number of arguments (given 0, expected 1))
irb(main):034:0> 
```

File: enumerable_methods.rb [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    elsif object.is_a?(Hash)
+      my_each(object) do |key, value|
+        next if yield [key, value]
+
+        flag = false
+        break
+      end
+    end
+    flag
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):036:0> true_array = [1, true, 'hi', []]
=> [1, true, "hi", []]
irb(main):037:0> true_array.all?
=> true
irb(main):038:0> true_array.my_all?
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):38
        1: from enumerable_methods.rb:48:in `my_all?'
ArgumentError (wrong number of arguments (given 0, expected 1))
irb(main):039:0> 
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):046:0> array
=> [1, 7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):047:0> array[0] = 'word'
=> "word"
irb(main):048:0> array.all?(Integer)
=> false
irb(main):050:0> array.my_all?(Integer)
=> true
irb(main):051:0> 
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):053:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):054:0> words.all?(/o/)
=> false
irb(main):055:0> words.my_all?(/o/)
=> true
irb(main):056:0>
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):060:0> array
=> [7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):061:0> array.all?(3)
=> false
irb(main):062:0> array.my_all?(3)
=> true
irb(main):063:0> 
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):065:0> true_block = proc { |num| num <= 9 }
=> #<Proc:0x000055bf3ef1b578@(irb):65>
irb(main):066:0> array.all?(&true_block)
=> true
irb(main):067:0> array.my_all?(&true_block)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):67
        1: from enumerable_methods.rb:48:in `my_all?'
ArgumentError (wrong number of arguments (given 0, expected 1))
irb(main):068:0> 
```

File: enumerable_methods.rb [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    elsif object.is_a?(Hash)
+      my_each(object) do |key, value|
+        next unless yield [key, value]
+
+        flag = true
+        break
+      end
+    end
+    flag
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):070:0> true_array = [nil, false, true, []]
=> [nil, false, true, []]
irb(main):071:0> true_array.any?
=> true
irb(main):072:0> true_array.my_any?
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):72
        1: from enumerable_methods.rb:68:in `my_any?'
ArgumentError (wrong number of arguments (given 0, expected 1))
irb(main):073:0> 
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):075:0> array
=> [7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):076:0> array.any?(Integer)
=> true
irb(main):077:0> array.my_any?(Integer)
=> false
irb(main):078:0>
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):080:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):081:0> words.any?(/d/)
=> true
irb(main):082:0> words.my_any?(/d/)
=> false
irb(main):083:0>
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):086:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):087:0> words[0] = 'cat'
=> "cat"
irb(main):088:0> words.any?('cat')
=> true
irb(main):089:0> words.my_any?('cat')
=> false
irb(main):090:0>
```

File: enumerable_methods.rb [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    elsif object.is_a?(Hash)
+      my_each(object) do |key, value|
+        next unless yield [key, value]
+
+        flag = false
+        break
+      end
+    end
+    flag
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):094:0> true_block = proc { |num| num > 9 }
=> #<Proc:0x000055bf3ef124a0@(irb):94>
irb(main):095:0> array.none?(&true_block)
=> true
irb(main):096:0> array.my_none?(&true_block)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):96
        1: from enumerable_methods.rb:88:in `my_none?'
ArgumentError (wrong number of arguments (given 0, expected 1))
irb(main):097:0> 
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):099:0> false_array = [nil, false, nil, false]
=> [nil, false, nil, false]
irb(main):100:0> false_array.none?
=> true
irb(main):101:0> false_array.my_none?
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):101
        1: from enumerable_methods.rb:88:in `my_none?'
ArgumentError (wrong number of arguments (given 0, expected 1))
irb(main):102:0> 
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):104:0> array
=> [7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):105:0> array.none?
=> false
irb(main):106:0> array.my_none?
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):106
        1: from enumerable_methods.rb:88:in `my_none?'
ArgumentError (wrong number of arguments (given 0, expected 1))
irb(main):107:0>
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):109:0> words.none?(/d/)
=> false
irb(main):110:0> words.my_none?(/d/)
=> true
irb(main):111:0> words
=> ["cat", "door", "rod", "blade"]
irb(main):112:0>
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):119:0> words
=> ["cat", "door", "rod", "blade"]
irb(main):120:0> words[0] = 5
=> 5
irb(main):121:0> words.none?(5)
=> false
irb(main):122:0> words.my_none?(5)
=> true
irb(main):123:0>
```

File: enumerable_methods.rb [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+      end
+    elsif object.is_a?(Hash)
+      my_each(object) do |key, value|
+        next unless yield [key, value]
+
+        count += 1
+      end
+    end
+    count
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):125:0> array
=> [7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):126:0> array.count
=> 99
irb(main):127:0> array.my_count
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):127
        1: from enumerable_methods.rb:108:in `my_count'
ArgumentError (wrong number of arguments (given 0, expected 1))
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):128:0> array.count(3)
=> 10
irb(main):129:0> array.my_count(3)
=> 0
irb(main):130:0> 
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):132:0> block = proc { |num| num < 9/2 } 
=> #<Proc:0x000055bf3edc3a90@(irb):132>
irb(main):133:0> array.count(&block)
=> 41
irb(main):134:0> array.my_count(&block)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):134
        1: from enumerable_methods.rb:108:in `my_count'
ArgumentError (wrong number of arguments (given 0, expected 1))
irb(main):135:0> 
```

File: enumerable_methods.rb [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+    my_each(object) do |x|
+      if !proc
+        new_object.push(yield x)
+      else
+        new_object.push(proc.call(x))
+      end
+    end
+    new_object
+  end
+
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):137:0> array
=> [7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):138:0> array.map
=> #<Enumerator: [7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]:map>
irb(main):139:0> array.my_map
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):139
        1: from enumerable_methods.rb:126:in `my_map'
ArgumentError (wrong number of arguments (given 0, expected 1..2))
irb(main):140:0>
```

File: enumerable_methods.rb [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+  def my_inject(object, acc = 0)
+    my_each(object) do |x|
+      acc = yield acc, x
+    end
+    acc
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):142:0> operation = proc { |sum, n| sum + n }
=> #<Proc:0x000055bf3ef1bdc0@(irb):142>
irb(main):143:0> array.inject(&operation)
=> 410
irb(main):144:0> array.my_inject(&operation)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):144
        1: from enumerable_methods.rb:138:in `my_inject'
ArgumentError (wrong number of arguments (given 0, expected 1..2))
irb(main):145:0> array
=> [7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):146:0> 
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):145:0> array
=> [7, 7, 0, 1, 1, 4, 7, 1, 6, 7, 6, 4, 6, 4, 1, 1, 3, 3, 6, 5, 5, 7, 1, 5, 0, 8, 1, 8, 3, 8, 0, 4, 2, 8, 2, 4, 7, 5, 1, 2, 3, 7, 7, 2, 1, 0, 6, 6, 3, 5, 4, 2, 8, 1, 3, 2, 4, 0, 5, 1, 3, 2, 4, 5, 6, 7, 5, 6, 2, 5, 7, 8, 2, 5, 8, 6, 6, 5, 4, 5, 4, 3, 2, 7, 2, 8, 8, 3, 3, 5, 1, 8, 5, 1, 0, 4, 0, 6, 8]
irb(main):146:0> 
irb(main):147:0> 
irb(main):148:0> array.inject(:+)
=> 410
irb(main):149:0> array.my_inject(:+)
=> 0
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello @jstiven01 ,

Thank you for the job well done. However, there are some bugs to fix.
Please, read the comments below to learn more about the bugs.

Kindly make the changes and submit a new code review request.
Thanks

**Review by Ispirett — APPROVED** [review]
Great work @jstiven01  :+1:  :100:  you can merge now 

Please remember to create proper PR messages.

**Pull request merged**